src/cljc/rems/common/util.cljc

```diff
@@ -19,6 +19,100 @@
   (is (= nil
          (re-matches +phone-number-regex+ "+35845000000000000000000000000000"))))
 
+;; regex from https://stackoverflow.com/questions/5284147/validating-ipv4-addresses-with-regexp
+(def +valid-ip-address-regex+ #"(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$")
+
+(def +valid-ip-address-regex-version-six+ #"(([0-9]|[a-f]){1,4}(\:)){7}(([0-9]|[a-f]){1,4})$")
+
+;; https://stackoverflow.com/questions/2814002/private-ip-address-identifier-in-regular-expression
+;; for now, this test should test for the following private IPv4 address pattaerns
+;; 0.x.x.x
+;; 10.x.x.x - Private network, local communications
+;; 100.64.x.x–100.127.x.x - Private network, communications between a service provider and its subscribers
+;; 127.x.x.x - host, loopback addresses to the local host
+;; 169.254.x.x - Subnet, Used for link-local addresses[7] between two hosts on a single link when no IP address is otherwise specified
+;; 172.16.x.x – 172.31.x.x - Private network, local communications within a private network
+;; 192.0.0.x - Private network, IETF Protocol Assignments
+;; 192.0.2.x - Documentation, Assigned as TEST-NET-1, documentation and examples.
+;; 192.168.x.x - Private network, local communications within a private network
+;; 198.18.x.x - 198.19.x.x - Private network, benchmark testing of inter-network communications between two separate subnets
+;; 198.51.100.x - Documentation, Assigned as TEST-NET-2, documentation and examples.
+;; 203.0.113.x - Documentation, Assigned as TEST-NET-3, documentation and examples.
+;; 240.x.x.x–255.x.x.x - multicast
+(def +reserved-ip-address-range-regex+
+  (re-pattern
+   (str
+    "((0|10|127|224?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((100?)\\.([6-9][4-9]|1[1-2][1-7]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((169?)\\.(254?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((172?)\\.(1[6-9]|2[0-9]|31?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((192?)\\.(0?)\\.(0|2?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((192?)\\.(168?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((198?)\\.(1[8-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((198?)\\.(51?)\\.(100?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((203?)\\.(0?)\\.(113?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)|"
+    "((2[4-5][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)")))
```

Comment:
The regex-based approach seems a bit fragile. Did you consider parsing the IP addresses into vectors like `[169 254 13 14]` and using code to check the ranges? Alternatively, you could consider writing some sort of util to generate a regex from a pattern like `169.254.x.x`. The only difficult case is `100.64.x.x–100.127.x.x`, and even that could be fixe by just generating patterns `100.64.x.x`, `100.65.x.x`, ..., `100.127.x.x`

Reply:
One more thing that casme to my mind is that there should be no need to have things like `(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)` in these regexes. If a string is given to this regex, we know it has already passed `+valid-ip-address-regex+`, so we can just use patterns like `10\..*` or `169\.254\..*` here. Unless I'm missing something, that is.